src/manifold/src/boolean_result.cpp

```diff
@@ -145,6 +165,81 @@ void AddNewEdgeVerts(
   // the output vert index. When forward is false, all is reversed.
   const VecDH<int> &p1 = p1q2.Get(!forward);
   const VecDH<int> &q2 = p1q2.Get(forward);
+#if MANIFOLD_PAR == 'T' && __has_include(<tbb/tbb.h>)
+  // parallelize operations, requires concurrent_map so we can only enable this
+  // with tbb
+  if (p1q2.size() > 128) {
+    // ideally we should have 1 mutex per key, but 64 is enough to avoid
+    // contention for most of the cases
+    std::array<std::mutex, 128> mutexes;
+    static tbb::affinity_partitioner ap;
+    tbb::parallel_for(
+        tbb::blocked_range<int>(0, p1q2.size(), 32),
+        [&](const tbb::blocked_range<int> &range) {
+          for (int i = range.begin(); i != range.end(); i++) {
+            const int edgeP = p1[i];
+            const int faceQ = q2[i];
+            const int vert = v12R[i];
+            const int inclusion = i12[i];
+            Halfedge halfedge = halfedgeP[edgeP];
+
+            {
+              size_t hash = std::hash<int>()(edgeP) % mutexes.size();
+              EdgePos edgePos = {vert, 0.0f, inclusion < 0};
+              auto &edgePosP = edgesP[edgeP];
+              // each EdgePos need to be locked independently,
+              // as pushing elements into a vector may reallocate and
+              // invalidate the pointer
+              mutexes[hash].lock();
+              for (int j = 0; j < glm::abs(inclusion); ++j) {
+                edgePosP.push_back(edgePos);
+                ++edgePos.vert;
+              }
+              mutexes[hash].unlock();
+            }
```

Comment:
For readability, it might be nice to turn these into actual functions, especially if it allows us to show that they only affect a subset of the local variables here.